### ISSUE TITLE
Error in documentation for foreign build sandboxing

### ISSUE THREAD

**Issue by samoht** [ocaml/dune#3345]
https://dune.readthedocs.io/en/latest/foreign-code.html#foreign-sandboxing is recommending to write this dune file:

```
(data_only_dirs libfoo)

(rule
 (deps (source_tree libfoo))
 (targets libfoo.a dllfoo.so)
 (action (progn
          (chdir libfoo (run make))
          (copy libfoo/libfoo.a libfoo.a)
          (copy libfoo/libfoo.so dllfoo.so)))

(library
 (name bar)
 (foreign_archives foo))
```

Otherwise this doesn't work as the `copy` stanzas will fail because the files do not exist before the build ir run, and `dune` complains that it doens't know how to build these.

**Comment by diml** [comment]
Indeed. We need to add a mode to disable targets/deps inference. Directory targets (#3316) would also help here.

**Comment by robinbb** [comment]
This appears to have been fixed in #3456 (merged 2020-05-07). The current `doc/foreign-code.rst` example, under **Foreign Build Sandboxing**, includes the explicit `(copy ...)` actions you suggested (and adds `(no-infer)` for good measure). Closing as done.
